### PR TITLE
feat: add details to file structures to show test assets and types

### LINKED ISSUE #100: Document how test assets and types should be structured
A proposed structure, using elements as an example, is as follows:

```
Elements/
  index.ts
  types.ts
  myElement/
    myElement.ts
    myElement.spec.ts
    myElement.assets.ts
    myElement.types.ts
```

### PR DESCRIPTION
**Pull request by JQrdan**
Closes: #100

Signed-off-by: Jordan <jordan.tucker1@ibm.com>

### CODE REVIEW

**Review by matthew-chirgwin — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

File: client/Hooks/README.md [hunk 1]
```diff
@@ -13,16 +13,24 @@ For a given Hook `useFoo`, the expected files are as follows:
 ```
 Hooks/
   index.ts
+  types.ts
   useFoo/
     README.md
     Hook.ts
```
Comment (nictownsend) [inline client/Hooks/README.md]: UseFoo.*

File: client/Contexts/README.md [hunk 2]
```diff
@@ -15,16 +15,24 @@ For a given new Context `FeatureFlagContext`, the expected files are as follows:
 ```
 Contexts/
   index.ts
+  types.ts
   FeatureFlagContext/
     README.md
     Context.ts
+    Context.assets.ts
```
Comment (nictownsend) [inline client/Contexts/README.md]: FeatureFlagContext

**Review by nictownsend — CHANGES_REQUESTED** [review]
Minor changes

File: client/Hooks/README.md (hunk 1 above)
Comment (nictownsend) [reply client/Hooks/README.md]: Actually, we should name the hook `Foo`, and then it's called `useFoo` when exported?

**Review by matthew-chirgwin — CHANGES_REQUESTED** [review]
https://github.com/strimzi/strimzi-ui/blob/master/docs/Architecture.md#file-name-conventions also needs an update to reference the `types.ts` file

**Review by nictownsend — APPROVED** [review]

**Comment by nictownsend** [comment]
Approved - but https://github.com/strimzi/strimzi-ui/pull/104#discussion_r522856913 would be nice if you agree

**Comment by JQrdan** [comment]
> Approved - but [#104 (comment)](https://github.com/strimzi/strimzi-ui/pull/104#discussion_r522856913) would be nice if you agree

I've got mixed feelings about this, I'd rather keep the file name as `useFoo` but am open to changing it if others think it should change

File: client/Elements/README.md [hunk 3]
```diff
@@ -13,19 +13,29 @@ For a given Element component `Heading`, the expected files are as follows:
...
 
 - index.ts acts as a barrel file, exporting all public elements of this component/the components in the Elements directory
+- types.ts acts as a barrel file, exporting all the public types of each element
 - README.md is the readme for this component, detailing design choices and usage
 - View.ts is the view logic for this component
 - Styling.scss (_optional_) is the styling for this component
+- Heading.types.ts are all the types for this component
+- heading.feature is the feature definition to test this component
+- Heading.steps.ts are the steps for the feature file
+- HEading.assets.ts are the test assets for this component
```
Comment (pmuir) [inline client/Elements/README.md]: type on last line I think

**Review by pmuir — APPROVED** [review]

**Comment by matthew-chirgwin** [comment]
> > Approved - but [#104 (comment)](https://github.com/strimzi/strimzi-ui/pull/104#discussion_r522856913) would be nice if you agree
> 
> I've got mixed feelings about this, I'd rather keep the file name as `useFoo` but am open to changing it if others think it should change

I would vote to keep it as is. Just imagining the case of someone using the `useFoo` hook and wanting to find it in the codebase. You would hope they would realise `useFoo` lives in the `Foo` directory, but `useFoo` is explicit

**Review by dlabaj — APPROVED** [review]

**Review by matthew-chirgwin — APPROVED** [review]

**Pull request merged**